File: Code/BioMed-DI/IR-GS-Creation/DI2/DI2Datasets.cs
using System;
using System.IO;

namespace IR_GS_Creation
{
    class DI2Datasets
    {

        public static FileInfo DI1_dt = new FileInfo(Environment.CurrentDirectory + "/data/input/DI2/DI1_dt.tsv");

        public static FileInfo all_gene_disease_pmid_associations_dt(int fileNumber) 
        {

            return new FileInfo(Environment.CurrentDirectory + "/data/input/DI2/all_gene_disease_pmid_associations_" + fileNumber + "_dt.tsv");

        }

    }

}

File: Code/BioMed-DI/IdentityResolution/src/main/java/genes/IdentityResolution/solutions/DI2/DI2Datasets.java
package genes.IdentityResolution.solutions.DI2;

// java
import java.io.File;

// winter
import de.uni_mannheim.informatik.dws.winter.model.HashedDataSet;
import de.uni_mannheim.informatik.dws.winter.model.defaultmodel.Attribute;

// model
import genes.IdentityResolution.model.Gene.Gene;
import genes.IdentityResolution.model.Gene.GeneXMLReader;

public class DI2Datasets {


    public static HashedDataSet<Gene, Attribute> kaessmann() throws Exception {

        HashedDataSet<Gene, Attribute> ds = new HashedDataSet<>();

        new GeneXMLReader().loadFromXML(new File("data/input/DI2/DI1_dt.xml"), "/genes/gene", ds);

        return ds;

    }


    public static HashedDataSet<Gene, Attribute> all_gene_disease_pmid_associations(int i) throws Exception {

        HashedDataSet<Gene, Attribute> ds = new HashedDataSet<>();

        new GeneXMLReader().loadFromXML(new File("data/input/DI2/all_gene_disease_pmid_associations_" + i + "_dt.xml"), "/genes/gene", ds);

        return ds;

    }
 
    
}